projects/media-viewer/src/lib/icp/icp.service.ts

```diff
@@ -33,10 +27,11 @@ export class IcpDirective implements OnInit, OnDestroy {
               private readonly socketService: IcpUpdateService,
               private readonly presenterSubscriptions: IcpPresenterService,
               private readonly followerSubscriptions: IcpFollowerService,
-              private store: Store<IcpState>) {}
-
-  ngOnInit() {
-    this.subscription = this.toolbarEvents.icp.sessionLaunch.subscribe(() => this.launchSession());
+              private store: Store<IcpState>) {
+    this.store.pipe(select(fromIcpSelectors.getCaseId), filter(value => !!value)).subscribe(caseId => {
+      this.caseId = caseId;
+      this.subscription = this.toolbarEvents.icp.sessionLaunch.subscribe(() => this.launchSession());
```

Comment:
not sure we need to nest the two subscriptions, we can just check if caseId is set